idaes/models/unit_models/stream_scaler.py

```diff
@@ -0,0 +1,255 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+Unit model to adjust size of streams to represent, for example, a stream being split across several identical units,
+which are then all modeled as a single IDAES unit
+"""
+from enum import Enum
+from functools import partial
+
+from pyomo.environ import (
+    Block,
+    check_optimal_termination,
+    Param,
+    PositiveReals,
+    Reals,
+    RangeSet,
+    units as pyunits,
+    Var,
+)
```

Comment:
These pylint issues need to be addressed for the tests to pass.

Reply:
On it!